tests/unit/test_charm.py

```diff
@@ -20,6 +26,16 @@ def test_install_failure_blocked_status():
         assert state_out.unit_status == ops.BlockedStatus("unable to install etcd snap")
 
 
+def test_internal_user_creation():
+    ctx = testing.Context(EtcdOperatorCharm)
+    relation = testing.PeerRelation(id=1, endpoint=PEER_RELATION)
+
+    state_in = testing.State(relations={relation}, leader=True)
+    state_out = ctx.run(ctx.on.leader_elected(), state_in)
+    secret_out = state_out.get_secret(label=f"{PEER_RELATION}.{APP_NAME}.app")
+    assert secret_out.latest_content.get(f"{INTERNAL_USER}-password")
```

Comment:
nit: `assert secret_out.latest_content`

Reply:
This will not work as we get back the whole dict.